tests/storage/guestfs.go

```diff
@@ -80,36 +84,41 @@ var _ = SIGDescribe("[rfe_id:6364][[Serial]Guestfs", func() {
 			Expect(guestfsCmd.Execute()).ToNot(HaveOccurred())
 		}()
 		// Waiting until the libguestfs pod is running
-		Eventually(func() corev1.PodPhase {
+		Eventually(func() bool {
 			pod, _ := virtClient.CoreV1().Pods(util.NamespaceTestDefault).Get(context.Background(), podName, metav1.GetOptions{})
-			return pod.Status.Phase
-		}, 90*time.Second, 2*time.Second).Should(Equal(k8sv1.PodRunning))
+			ready := false
+			for _, status := range pod.Status.ContainerStatuses {
+				if status.State.Running != nil {
+					return true
+				}
+			}
+			return ready
+
+		}, 90*time.Second, 2*time.Second).Should(BeTrue())
 		// Verify that the appliance has been extracted before running any tests
 		// We check that all files are present and tar is not running
 		Eventually(func() bool {
 			output, _, err := execCommandLibguestfsPod(podName, []string{"ls", "-m", "/usr/local/lib/guestfs/appliance"})
-			Expect(err).ToNot(HaveOccurred())
+			if err != nil {
+				return false
+			}
 			if !strings.Contains(output, APPLIANCE_FILES) {
 				return false
 			}
-			output, _, err = execCommandLibguestfsPod(podName, []string{"ps", "-e", "-o", "comm="})
+			output, _, err = execCommandLibguestfsPod(podName, []string{"du", "-b", "/usr/local/lib/guestfs/appliance/root"})
+			if err != nil {
+				return false
+			}
+			size, err := strconv.Atoi(strings.Trim(strings.Split(output, "\t")[0], " "))
 			Expect(err).ToNot(HaveOccurred())
-			if strings.Contains(output, "tar") {
```

Comment:
are we ok with dropping the tar check?
iiuc it was tackling a different flake:
https://github.com/kubevirt/kubevirt/pull/6474

Reply:
@alicefr can you maybe just add another line to the libguestfs script to indicate that you are done?

Something like this:

```
#!/bin/bash -xe

DIR=/usr/local/lib/guestfs
LIBGUEST_APPLIANCE=${DIR}/downloaded
tar -Jxf ${LIBGUEST_APPLIANCE} -C ${DIR}

touch ${DIR}/done

/bin/bash
```

Reply:
@akalenyu I'm checking the size with du. So the root needs to be completely extracted. But I like @rmohr suggestion :).

Reply:
right

Reply:
> Also I am missing context a little, but in general if it matters that a task is done and we have to clearly see it from outside, consider doing the task in an init container. That container can exit and we can clearly see the success on the api.

@rmohr the only issue I see it's that I need another container image for the init container. It needs at least to have the tar command.

Reply:
@alicefr you can use the same container, but if it does not fit well, just don't do it. Was just a thought.

Reply:
@rmohr the init container is actually the right way of doing this, It could even avoid the done file. Maybe I can do it in a second PR. If we merge this quickly it might solve some of the flaky tests

Reply:
:+1: